### PR TITLE
Fix fix asn tags... again

### PR DESCRIPTION
**Pull request by user12986714**
We need to first delete tags before adding new ones to prevent duplicated tags error.

### CODE REVIEW

**Comment by thesecretmaster** [comment]
What problem is this solving? In what case is there a duplicated tags error?

**Comment by user12986714** [comment]
POST https://metasmoke.erwaysoftware.com/domains/3/fix_asn , gives 
```
ActiveRecord::RecordNotUnique in SpamDomainsController#fix_asn_tags
Mysql2::Error: Duplicate entry '323-3' for key 'PRIMARY': INSERT INTO `domain_tags_spam_domains` (`domain_tag_id`, `spam_domain_id`) VALUES (323, 3)
```

File: app/models/spam_domain.rb [hunk 1]
```diff
@@ -20,17 +20,15 @@ def fix_asn_tags
...
-    prev_domain_tags = domain_tags.where(special: true).select(:name).map(&:name).select { |dt| dt.start_with?('AS-') }
+    domain_tags.where(special: true).select(:name).map(&:name).select { |dt| dt.start_with?('AS-') }.each do |dtn|
+      domain_tags.delete(DomainTag.find_by(name: dtn))
+    end
     asn.split.each do |as|
       desc = `dig +short AS#{as}.asn.cymru.com TXT`.strip.tr('"', '').split('|')[-1]&.strip
       tag = DomainTag.find_or_create_by(name: "AS-#{as}", special: true)
-      prev_domain_tags -= ["AS-#{as}"]
       tag.update(description: "Domains under the Autonomous System Number #{as} - #{desc}.") unless tag.description.present?
       domain_tags << tag
```
Comment (thesecretmaster) [inline app/models/spam_domain.rb:30]: I'd prefer to fix this by sticking an `unless domain_tags.include? tag` on this line.

Comment (thesecretmaster) [reply app/models/spam_domain.rb:30]: It's just going to be slightly better than clearing all the AS tags then re-creating them.

**Pull request merged**